### PR TITLE
Implement basic file based storage

### PR DESCRIPTION
**Pull request by IanDBird**
As discussed with @koalaparadise, it was not ideal that we were using `SharedPreferences`, especially when we were dumping JSON into it. There was no benefit to having this (very tight) Android dependency. For now, this migrates to just a basic File that still leverages our existing JSON serialisation support.

### CODE REVIEW

File: sdk/src/main/java/com/uid2/storage/FileStorageManager.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+internal class FileStorageManager(
+    private val identityFile: File,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+): StorageManager {
+
+    constructor(context: Context): this(File(context.filesDir, FILE_IDENTITY))
+
+    override suspend fun saveIdentity(identity: UID2Identity, status: IdentityStatus) = withContext(ioDispatcher) {
+        return@withContext runCatching {
+            OutputStreamWriter(identityFile.outputStream()).use { writer ->
```
Comment (koalaparadise) [inline sdk/src/main/java/com/uid2/storage/FileStorageManager.kt:27]: The idiomatic Kotlin way to do this is:

```
identityFile.bufferedWriter().use { out -> out.write(jsonString) }
```

File: sdk/src/main/java/com/uid2/storage/FileStorageManager.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ */
+internal class FileStorageManager(
+    private val identityFile: File,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+): StorageManager {
+
+    constructor(context: Context): this(File(context.filesDir, FILE_IDENTITY))
+
+    override suspend fun saveIdentity(identity: UID2Identity, status: IdentityStatus) = withContext(ioDispatcher) {
+        return@withContext runCatching {
```
Comment (koalaparadise) [inline sdk/src/main/java/com/uid2/storage/FileStorageManager.kt:26]: No need for this

File: sdk/src/main/java/com/uid2/storage/FileStorageManager.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+                writer.write(identityJson.toString(0))
+                return@runCatching true
+            }
+        }.getOrDefault(false)
+    }
+
+    override suspend fun loadIdentity(): Pair<UID2Identity?, IdentityStatus> = withContext(ioDispatcher) {
+        return@withContext runCatching {
+            InputStreamReader(identityFile.inputStream()).use { reader ->
```
Comment (koalaparadise) [inline sdk/src/main/java/com/uid2/storage/FileStorageManager.kt:41]: Similarly, idiomatic way would be:

```
val jsonString = identityFile.readText(Charsets.UTF_8)
```

File: sdk/src/main/java/com/uid2/storage/FileStorageManager.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                return@runCatching Pair(
+                    UID2Identity.fromJson(identityJson),
+                    IdentityStatus.fromValue(identityJson.getInt(KEY_STATUS))
+                )
+            }
+        }.getOrDefault(Pair(null, NO_IDENTITY))
+    }
+
+    override suspend fun clear() = withContext(ioDispatcher) {
+        return@withContext identityFile.delete()
```
Comment (koalaparadise) [inline sdk/src/main/java/com/uid2/storage/FileStorageManager.kt:53]: No need for this

**Review by koalaparadise — COMMENTED** [review]
Looks good overall! Kotlin gives you some more shortcuts.

File: sdk/src/main/java/com/uid2/storage/FileStorageManager.kt (hunk 1 above)
Comment (IanDBird) [reply sdk/src/main/java/com/uid2/storage/FileStorageManager.kt]: Sounds good, will update!

File: sdk/src/main/java/com/uid2/storage/FileStorageManager.kt (hunk 3 above)
Comment (IanDBird) [reply sdk/src/main/java/com/uid2/storage/FileStorageManager.kt]: Sure. If I've being explicit with the `Charset` during the read, I'll do that with the write too.

**Review by koalaparadise — APPROVED** [review]

**Pull request merged**